ga4gh/datamodel/variants.py

```diff
@@ -547,3 +596,160 @@ def buildMetadata(
+        coordLen = pos.split('/')
+        allLoc = self._createGaAlleleLocation()
+        allLoc.overlapStart = int(coordLen[0])
+        return allLoc
+
+    def convertLocationHgvsC(self, hgvsc):
+        if hgvsc == '':
+            return None
+        match = re.match("c.(\d+)(\D+)>(\D+)", hgvsc)
+        if match:
+            pos = int(match.group(1))
+            if pos > 0:
+                allLoc = self._createGaAlleleLocation()
+                allLoc.overlapStart = pos
+                allLoc.referenceSequence = match.group(2)
+                allLoc.alternateSequence = match.group(3)
+                return allLoc
+        return None
+
+    def convertLocationHgvsP(self, hgvsp):
+        if hgvsp == '':
+            return None
+        match = re.match("p.(\D+)(\d+)(\D+)", hgvsp)
+        if match is not None:
+            allLoc = self._createGaAlleleLocation()
+            allLoc.referenceSequence = match.group(1)
+            allLoc.overlapStart = int(match.group(2))
+            allLoc.alternateSequence = match.group(3)
+            return allLoc
+        return None
+
+    def convertTranscriptEffect(self, ann):
+        effect = self._createGaTranscriptEffect()
+        (alt, effects, impact, geneName, geneId, featureType,
+            featureId, trBiotype, rank, hgvsC, hgvsP, cdnaPos,
+            cdsPos, protPos, distance, errsWarns) = ann.split('|')
+        effect.alternateBases = alt
+        effect.effects = self.convertSeqOntology(effects)
+        effect.impact = impact
+        effect.featureId = featureId
+        effect.HGVSc = hgvsC
+        effect.HGVSp = hgvsP
+        if hgvsP != '':
+            effect.proteinLocation = self.convertLocationHgvsP(hgvsP)
+        if effect.proteinLocation is None:
+            effect.proteinLocation = self.convertLocation(protPos)
+        if hgvsC != '':
+            effect.CDSLocation = self.convertLocationHgvsC(hgvsC)
+        if effect.CDSLocation is None:
+            effect.CDSLocation = self.convertLocation(cdnaPos)
+        effect.cDNALocation = self.convertLocation(cdnaPos)
+        effect.id = self.getTranscriptEffectId(effect)
+        return effect
+
+    def convertSeqOntology(self, seqOntStr):
+        seqOntTerms = seqOntStr.split('&')
+        soTerms = []
+        for soName in seqOntTerms:
+            so = self._createGaOntologyTermSo()
+            so.name = soName
+            if self._sequenceOntology is not None:
+                so.id = self._sequenceOntology.getId(soName)
+            soTerms.append(so)
+            # TODO We must fill the ontology ID based on the SO name
+        return soTerms
+
+    def convertVariantAnnotation(self, record):
+        """
+        Converts the specified record into a GA4GH Annotaiton object.
+        """
+        # First we need the Variant
+        variant = self.convertVariant(record, None)
+        # Create annotaiton
+        annotation = self._createGaVariantAnnotation()
+        annotation.variant = variant
```

Comment:
Do we really embed the variant in the annotation object? I just see variantId in the schema?